pkg/domain/infra/abi/secrets.go

```diff
@@ -166,17 +166,10 @@ func (ic *ContainerEngine) SecretRm(ctx context.Context, nameOrIDs []string, opt
 	}
 	for _, nameOrID := range toRemove {
 		deletedID, err := manager.Delete(nameOrID)
-		if err == nil || strings.Contains(err.Error(), "no such secret") {
-			if !options.Ignore {
-				reports = append(reports, &entities.SecretRmReport{
-					Err: err,
-					ID:  deletedID,
-				})
-			}
+		if options.Ignore && errors.Is(err, secrets.ErrNoSuchSecret) {
 			continue
-		} else {
-			return nil, err
 		}
+		reports = append(reports, &entities.SecretRmReport{Err: err, ID: deletedID})
```

Comment:
This change not only changes the behavior on the `not found` but also continues on error to the next item (while adding the error to the report). Is this the behavior you intended to have?

Reply:
> Is this the behavior you intended to have?

Yes, that how other endpoints (usually) behave.  An error should only be returned on fatal ones that would prevent processing other objects.